### ISSUE TITLE
SVG output, text is not centered vertically

### ISSUE THREAD

**Issue by neoteknic** [laravolt/avatar#66]
With SVG output, text is not centered vertically.
Chrome : not well centered
Edge : text is on top

Solution : correct default value + add parameters to give an offset X, Y to the text.

version : 2.1.0

**Comment by neoteknic** [comment]
+ suggestion : is it possible to vectorize text instead of output it in SVG ?

**Comment by uyab** [comment]
Hi @neoteknic, sorry I little bit busy in recent weeks. Any help welcome for this issue :)

**Comment by neoteknic** [comment]
![image](https://user-images.githubusercontent.com/1809652/52999801-07c17a80-3427-11e9-97b3-b7ff6c8ae6c5.png)
Maybe a font issue, look at this image on Google chrome windows 10 x64.

**Comment by neoteknic** [comment]
![image](https://user-images.githubusercontent.com/1809652/53010401-20895a80-343e-11e9-9f83-f7d3afaef686.png)

Chrome ; Firefox ; Edge 
Browser issue ? but possible to fire in Avatar.php toSvg() maybe.

**Comment by fannyfan414** [comment]
I extend Avatar class and in method toSvg add 4px to "y" coordinate.
This line 

> $svg .= '<text x="' . $center . '" y="' . ($center + 4);

With this modification content is now aligned correctly

```
<?php
/**
 * Created by PhpStorm.
 * User: Workstation
 * Date: 11.09.2019
 * Time: 23:21
 */

namespace App\Services\AvatarGenerator;


use Laravolt\Avatar\Avatar;

class AvatarGenerator extends Avatar
{
    public function toSvg()
    {
        $this->buildInitial();

        $x = $y = $this->borderSize / 2;
        $width = $height = $this->width - $this->borderSize;
        $radius = ($this->width - $this->borderSize) / 2;
        $center = $this->width / 2;

        $svg = '<svg version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" xml:space="preserve" width="' . $this->width . '" height="' . $this->height . '">';

        if ($this->shape == 'square') {
            $svg .= '<rect x="' . $x
                . '" y="' . $y
                . '" width="' . $width . '" height="' . $height
                . '" stroke="' . $this->borderColor
                . '" stroke-width="' . $this->borderSize
                . '" fill="' . $this->background . '" />';
        } elseif ($this->shape == 'circle') {
            $svg .= '<circle cx="' . $center
                . '" cy="' . $center
                . '" r="' . $radius
                . '" stroke="' . $this->borderColor
                . '" stroke-width="' . $this->borderSize
                . '" fill="' . $this->background . '" />';
        }

        $svg .= '<text x="' . $center . '" y="' . ($center + 4);
        $svg .= '" font-size="' . $this->fontSize;

        if ($this->fontFamily) {
            $svg .= '" font-family="' . $this->fontFamily;
        }

        $svg .= '" fill="' . $this->foreground . '" alignment-baseline="middle" text-anchor="middle" dominant-baseline="central">';
        $svg .= $this->getInitial();
        $svg .= '</text>';

        $svg .= '</svg>';

        return $svg;
    }

}
```

**Comment by uyab** [comment]
Just found https://github.com/meyfa/php-svg, will try to refactor the svg part using it.

**Comment by danpalmieri** [comment]
Poor (but fast 😶) workaround:
`{!! Str::replaceFirst(' y="50"', ' y="54"', Avatar::create('Susilo Bambang Yudhoyono')->toSvg()) !!}`

**Comment by uyab** [comment]
Fixed in 4.1.4
